airupnp/src/airupnp.c

```diff
@@ -819,6 +819,11 @@ static void *UpdateThread(void *args)
 				ModelNumber = XMLGetFirstDocumentItem(DescDoc, "modelNumber", true);
 				UDN = XMLGetFirstDocumentItem(DescDoc, "UDN", true);
 
+				// Check for null returns
+				if (ModelName == NULL || ModelNumber == NULL || UDN == NULL) {
+				  goto cleanup;
+				}
+
```

Comment:
I don't think we should not create the player if one of the returned item is NULL. We should let isExcluded do the check and accept validate the device if information is missing

Reply:
Yes, I saw how you'd approached it. Looks better.